circles-timeline/index.html

```diff
@@ -1,139 +1,187 @@
+    
 </body>
-</html>
\ No newline at end of file
+
+<script type="text/javascript">
+
+const normalColours = gChartcolour.categorical_bar;
+const popColours = gChartcolour.lineSocial;
+
+const webMargin = {
+    left: 20,
+    right: 30,
+    top: 130,
+    bottom: 30,
+}
+const printMargin = {
+    left: 10,
+    right: 20,
+    top: 100,
+    bottom: 30,
+}
+const socialMargin = {
+    left: 30,
+    right: 50,
+    top: 200,
+    bottom: 0,
+}
+const videoMargin = {
+    left: 100,
+    right: 50,
+    top: 300,
+    bottom: 0,
+}
+
+const frame = {
+    web: gChartframe.webFrame()
+        .margin(webMargin)
+        .width(600)
+        .height(640),
+
+    print: gChartframe.printFrame()
+        .margin(printMargin)
+        .width(221)   //note, this value is in mm for the print frame
+        .height(220),   //note, this value is in mm for the print frame
+
+    social: gChartframe.socialFrame()
+        .margin(socialMargin)
+        .width(560)
+        .height(720)
+        .title('Slightly shorter title perhaps?'),
+
+    video: gChartframe.videoFrame()
+        .margin(videoMargin)
+        .width(1920)
+        .height(1080)
+};
+
+d3.selectAll('.framed')
+    .each(function(){
+        const figure = d3.select(this);
+        figure.select('svg')
+            .call(frame[figure.node().dataset.frame]);
+    });
+
+//sort data 
+const dataSorter = function(a,b){
+    return a-b;
+};
+
+const formatId = function(name){
+    return name.replace(/\s+/g, '').toLowerCase();
+};
+
+d3.csv('data.csv',function(error, data){
+
+    if(error){ 
+        throw new Error("There is a problem the data entered into Circles Timeline chart: `${error}");
+    }
+    //parse the data 
+    const dates = [];
+    const values = [];  
+    data.sort(dataSorter);
+    data.forEach( d => {
+        dates.push(d.date);
+        values.push(d.value);
+    });
+
+    //establish max data value
+    values.sort(dataSorter);
+    const maxValue = values[values.length - 1];
+
+    //establish range of dates
+    dates.sort(dataSorter);
+    const dateDomain=[dates[0],dates[dates.length-1]];
+
+    //roll up the data by category 
+
+    let dataCategories = data.map((dataObj, index) => { return dataObj.category})
+    let dataCategoriesDeDuped = dataCategories.filter((category, position) => { return dataCategories.indexOf(category) == position; })
+    let dataCategoriesAsObjects = dataCategoriesDeDuped.map((category) => { return {key: category, values: [] } } );
+    const catData = dataCategoriesAsObjects.map((item) => { 
+                                                let allCategoryData = data.filter( (d) => { 
+                                                    return d.category === item.key;
+                                                } );
+                                            item.values = allCategoryData;
+                                            return item;
+                                    });
```

Comment:
That seems fine. I should say, I'm not against `nest` per se but i've found it can lead to awkward code; I've ended up rewriting enough bits of my own code where I've used nest that I'm inclined to use it sparingly esp. for things that are only nested one level deep.